include/imu_ros2/accelgyrotemp_data_provider.h

```diff
@@ -25,17 +25,61 @@
 #include "imu_ros2/accelgyrotemp_data_provider_interface.h"
 #include "imu_ros2/iio_wrapper.h"
 
+/**
+ * \brief Class implementation for accel gyro temp data provider.
+ *
+ * This class implements  AccelGyroTempDataProviderInterface interface.
+ * This class provides data for the publisher.
+ * The type of data are AccelGyroTempData.
```

Comment:
Either use plural or single in both cases: "The type of data is" or "The types of data are"